File: C++/remove-linked-list-elements.cpp
/**
 *  * Definition for singly-linked list.
 *   * struct ListNode {
 *    *     int val;
 *     *     ListNode *next;
 *      *     ListNode(int x) : val(x), next(NULL) {}
 *       * };
 *        */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode *fake = new ListNode(0);
        ListNode *prev = fake, *curr = head;
        fake->next = head;
        
        while(curr) {
            if(curr->val==val) {
                prev->next = curr->next;
                delete curr;
                curr = prev->next;
            }
            else {
                curr = curr->next;
                prev = prev->next;
            }
        }
        head = fake;
        head = head->next;
        delete fake;
        return head;
    }
};

File: Python/remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} head
    # @param {integer} val
    # @return {ListNode}
    def removeElements(self, head, val):
        
        prev = fake = ListNode(0)
        curr = head
        prev.next = curr
        
        while curr:
            if curr.val==val:
                curr = curr.next
                prev.next = curr
            else:
                curr = curr.next
                prev = prev.next
        
        return fake.next
